src/server/datachan.rs

```diff
@@ -267,6 +267,29 @@ where
 {
     // We introduce a block scope here to keep the lock on the session minimal. We basically copy the needed info
     // out and then unlock.
+
+    match socket.peer_addr() {
```

Comment:
OK for now, but in this method we are waiting twice now for the session lock, we could move this into a method and call it from the below block.

Reply:
moved things around, please check if this is better

Reply:
Good if it compiles :-)